### PR TITLE
Update CMake Instructions

### PR DESCRIPTION
**Pull request by skresearcher**
Make CMake flags flexible to compile as static/shared libraries for other platforms/architectures.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -31,29 +23,26 @@ endif ()
...
+message( "OBJ_CRYPTO_OSX is set to ${OBJ_CRYPTO_OSX}" )
+message( "OBJ_CRYPTO_BUILD_TESTS  is set to ${OBJ_CRYPTO_BUILD_TESTS}" ) 
+message( "OBJ_CRYPTO_BUILD_SHARED is set to ${OBJ_CRYPTO_BUILD_SHARED}" ) 
+message( "OBJ_CRYPTO_INSTALL is set to ${OBJ_CRYPTO_INSTALL}" )
 
-include(CTest)
+set(BUILD_SHARED_LIBS ${OBJ_CRYPTO_BUILD_SHARED})
 
+if ( OBJ_CRYPTO_OSX)
+    set( CMAKE_OSX_ARCHITECTURES "arm64;x86_64" CACHE STRING "OSX archs")
```
Comment (fluffy) [inline CMakeLists.txt:44]: I'm thinking we should take this out of the CAMKE and it can just be set on CLI to cmake

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -17,19 +16,9 @@ configure_file (
...
-add_library(
-  objCrypto
-)
-
-target_sources( objCrypto 
-  PRIVATE
-  objCrypto.cc
-  aes-ctr.cc 
-  aes-gcm.cc )
+add_library( objCrypto  aes-ctr.cc aes-ctr.h aes-gcm.cc aes-gcm.h objCrypto.cc )
```
Comment (fluffy) [inline src/CMakeLists.txt:19]: I'm worried that this loses the PRIVATE on the sources

**Review by fluffy — CHANGES_REQUESTED** [review]
Looks good - just to two small things

File: CMakeLists.txt (hunk 1 above)
Comment (skresearcher) [reply CMakeLists.txt]: Fixed.

File: src/CMakeLists.txt (hunk 2 above)
Comment (skresearcher) [reply src/CMakeLists.txt]: Fixed.

**Review by fluffy — APPROVED** [review]

**Pull request merged**